swarm/storage/mru/resource.go

```diff
@@ -840,20 +855,24 @@ func getAddressFromDataSig(datahash common.Hash, signature Signature) (common.Ad
 }
 
 // create an update chunk
-func newUpdateChunk(addr storage.Address, signature *Signature, period uint32, version uint32, name string, data []byte, datalength int) *storage.Chunk {
+func newUpdateChunk(mru *SignedResourceUpdate) *storage.Chunk {
 
-	// no signatures if no validator
-	var signaturelength int
-	if signature != nil {
-		signaturelength = signatureLength
+	if mru.rootAddr == nil || mru.metaHash == nil {
+		log.Warn("Call to newUpdateChunk with nil rootAddr or metaHash")
+		return nil
+	}
+	// a datalength field set to 0 means the content is a multihash
```

Comment:
Please remove obsolete comment

Reply:
gone. Multihash is now encoded explicitly as a flag.